internal/configuration/configuration.go

```diff
@@ -80,33 +83,39 @@ func (m *Manager) Update(message models.DeviceConfigurationMessage) error {
 	configurationEqual := reflect.DeepEqual(message.Configuration, m.deviceConfiguration.Configuration)
 	workloadsEqual := reflect.DeepEqual(message.Workloads, m.deviceConfiguration.Workloads)
 	log.Tracef("Initial config: [%v]; workloads equal: [%v]; configurationEqual: [%v]", m.IsInitialConfig(), workloadsEqual, configurationEqual)
+	var errors error
+
 	if m.IsInitialConfig() || !(configurationEqual && workloadsEqual) {
 		log.Tracef("Updating configuration: %v", message)
 		for _, observer := range m.observers {
 			err := observer.Update(message)
 			if err != nil {
-				return err
+				errors = multierror.Append(errors, fmt.Errorf("cannot update observer: %s", err))
+				return errors
```

Comment:
This will stop the execution on first error; probably should be moved outside of the for-loop